### PR TITLE
Add PyTorch-native K-quant pass

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes

Add `KQuant` pass under `olive.passes.pytorch.kquant` that implements the ggml K-quant weight-only search in PyTorch. A unified `_kquant_search` driver covers both variants:

- Asymmetric (analogue of ggml's `make_qkx2_quants` used by Q2_K/Q4_K/Q5_K): tracks per-group `(min, max)`, refines `(scale, offset)` via 2-D LSQ over a sweep of perturbed `iscale` factors.
- Symmetric (analogue of ggml's `make_qx_quants` used by Q3_K/Q6_K): uses `max|x|` as normalizer, zero point fixed at `midq`, refines scale via 1-D LSQ `sumlx/suml2`.

Outputs `(scales, zero_points)` shaped to match `WeightQuantizer.find_qparams`, with the search parameterized by `(maxq, minq)` pulled directly from each module's `WeightQuantizer` (via `get_maxq_minq`) so the algorithm and `finalize` agree on the integer range. Uses the shared `prepare_model`/`finalize` plumbing so embeddings and `lm_head` can be quantized and retied like the other PyTorch quant passes (RTN, GPTQ).

## Checklist before requesting a review
- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [x] Lint and apply fixes to your code by running `lintrunner -a`
- [x] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

Release note: New `KQuant` PyTorch pass for weight-only K-quant quantization (asymmetric + symmetric, 2/4/8-bit). `Rtn` and `KQuant` now also advertise `uint2`/`int2` precisions.

### CODE REVIEW

File: olive/passes/pytorch/kquant.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+            normalizer,
+            rmin,
+            l_min,
+            l_max,
+            initial_factor=float(qrange),
+            candidate_factors=candidate_factors,
+            refine_fn=_refine_asymmetric,
+        )
+        zero_point = torch.clamp(torch.round(float(minq) - offset / scale), l_min, l_max).to(torch.int32)
+
```
[the remark this answers is not in the document]
Comment (jambayk) [reply olive/passes/pytorch/kquant.py:225]: Good catch — fixed in a57b9df5. Asymmetric now clamps rmin<=0 / rmax>=0 and falls back to (-1, 1) for all-zero groups, mirroring `WeightQuantizer.find_qparams` and ggml's `make_qkx2_quants` `if (min > 0) min = 0` guard. Symmetric replaces a zero normalizer with 1. Added `test_kquant_find_qparams_handles_constant_groups` covering 0 / +100 / -7.5 across both modes.

File: olive/passes/pytorch/kquant.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    return scales, zero_points
+
+
+class KQuant(Pass):
+    """K-quant weight-only quantization (PyTorch-native).
+
+    Per-group weight quantization using the iterative weighted-least-squares
+    search from llama.cpp's ggml k-quants. Supports both asymmetric (scale and
+    zero point) and symmetric (scale only) variants for 4- and 8-bit weights of
+    ``nn.Linear`` and ``nn.Embedding`` modules.
```
[the remark this answers is not in the document]
Comment (jambayk) [reply olive/passes/pytorch/kquant.py]: Fixed in a57b9df5 — docstring now mentions 2/4/8-bit.

File: olive/passes/pytorch/kquant.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+        cls,
+        config: type[BasePassConfig],
+        accelerator_spec: AcceleratorSpec,
+    ) -> bool:
+        if not super().validate_config(config, accelerator_spec):
+            return False
+
+        if config.group_size <= 0 and config.group_size != -1:
+            logger.info("group_size must be -1 or greater than 0")
+            return False
```
[the remark this answers is not in the document]
Comment (jambayk) [reply olive/passes/pytorch/kquant.py:262]: Declined — this is a pre-existing gap shared by `Rtn` and `Gptq` (neither validates the `>=16 and power of 2` constraint that `QuantLinear` enforces). Worth adding, but it belongs in the shared `get_quantizer_config` / base pass validation, not just here. Out of scope for this PR.

**Review by xiaoyu-work — APPROVED** [review]
